File: api/src/main/java/club/yunzhi/workhome/entity/Student.java
package club.yunzhi.workhome.entity;

import com.fasterxml.jackson.annotation.JsonView;

import javax.persistence.*;

/**
 * 学生
 *
 * @author yunzhi
 */
@Entity
public class Student extends AbstractEntity {
    private String name;
    @Column(nullable = false)
    private String no;

    @OneToOne
    @JoinColumn(nullable = false)
    @JsonView(UserJsonView.class)
    private User user;

    private Integer totalScore;

    private Integer averageScore;

    public Student() {
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getNo() {
        return no;
    }

    public void setNo(String no) {
        this.no = no;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public Integer getTotalScore() {
        return totalScore;
    }

    public void setTotalScore(Integer totalScore) {
        this.totalScore = totalScore;
    }

    public Integer getAverageScore() {
        return averageScore;
    }

    public void setAverageScore(Integer averageScore) {
        this.averageScore = averageScore;
    }

    public interface UserJsonView {
    }
}

File: workHomeWebApp/src/app/common/student.ts
import {User} from './user';

/**
 * 学生
 */

export class Student {

  id: number;

  name: string;

  no: string;

  totalScore: number;

  averageScore: number;

  user = new User();

  constructor(params?: {id?: number; name?: string; no?: string; user?: User; totalScore?: number; averageScore?: number}) {
    if (!params) {
      return;
    }
    this.id = params.id ? params.id : null;
    this.name = params.name ? params.name : null;
    this.no = params.no ? params.no : null;
    this.totalScore = params.totalScore ? params.totalScore : null;
    this.averageScore = params.averageScore ? params.averageScore : null;
    this.user = params.user ? params.user : new User();

  }
}
